cmd/testpeers.go

```diff
@@ -63,7 +108,145 @@ func supportedSelfTestCases() map[testCaseName]func(context.Context, *testPeersC
 	}
 }
 
-func runTestPeers(ctx context.Context, w io.Writer, cfg testPeersConfig) (err error) {
+func startTCPNode(ctx context.Context, cfg testPeersConfig) (host.Host, func(), error) {
+	var peers []p2p.Peer
+	for i, e := range cfg.ENRs {
+		record, err := enr.Parse(e)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "decode enr", z.Str("enr", e))
+		}
+
+		p, err := p2p.NewPeerFromENR(record, i)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		peers = append(peers, p)
+	}
+
+	key, err := p2p.LoadPrivKey(cfg.DataDir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	r, err := enr.New(key)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mePeer, err := p2p.NewPeerFromENR(r, len(cfg.ENRs))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Info(ctx, "Self p2p name resolved", z.Any("name", mePeer.Name))
+
+	peers = append(peers, mePeer)
+
+	allENRs := cfg.ENRs
+	allENRs = append(allENRs, r.String())
+	slices.Sort(allENRs)
+	allENRsString := strings.Join(allENRs, ",")
+	allENRsHash := sha256.Sum256([]byte(allENRsString))
+
+	return setupP2P(ctx, key, cfg.P2P, peers, allENRsHash[:])
+}
+
+func setupP2P(ctx context.Context, key *k1.PrivateKey, cfg p2p.Config, peers []p2p.Peer, enrsHash []byte) (host.Host, func(), error) {
```

Comment:
I wonder if we could export the setupP2P functions from `run` in a generic package and reuse everything... 

Just a random thought, we could explore this in the future.

Reply:
Yap, probably we can make it a bit more generic if we want to export it.

Reply:
We can ticket that as a low-prio ticket for the future.